src/createAsyncThunk.ts

```diff
@@ -0,0 +1,183 @@
+import { Dispatch } from 'redux'
+import nanoid from 'nanoid'
+import { createAction } from './createAction'
+
+export type BaseThunkAPI<S, E, D extends Dispatch = Dispatch> = {
+  dispatch: D
+  getState: () => S
+  extra: E
+  requestId: string
+  signal: AbortSignal
+}
+
+/**
+ * @alpha
+ */
+export interface SerializedError {
+  name?: string
+  message?: string
+  stack?: string
+  code?: string
+}
+
+const commonProperties: (keyof SerializedError)[] = [
+  'name',
+  'message',
+  'stack',
+  'code'
+]
+
+// Reworked from https://github.com/sindresorhus/serialize-error
+export const miniSerializeError = (value: any): any => {
+  if (typeof value === 'object' && value !== null) {
+    const simpleError: SerializedError = {}
+    for (const property of commonProperties) {
+      if (typeof value[property] === 'string') {
+        simpleError[property] = value[property]
+      }
+    }
+
+    return simpleError
+  }
+
+  return value
+}
+
+/**
+ *
+ * @param type
+ * @param payloadCreator
+ *
+ * @alpha
+ */
+export function createAsyncThunk<
```

Comment:
As far as I can tell you could make this whole parameter list into an interface.

Even though i haven't tried it, I believe you can even make certain properties on the _named property_ generic required. 

i.e.

```ts
interface createAsyncThunkI {
  Returned: unknown, // this is required
  ThunkArg?: void
  ...
}

export function createAsyncThunk<G extends createAsyncThunkI = createAsyncThunkI>() {}
```

Reply:
Right, that's exactly what I'm asking for help on figuring out :)

Reply:
Playing around with this locally now atm.  Lemme see if I can use this tiny snippet as inspiration...